In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('ingredientes_estandarizados.csv')

In [3]:
df = data.copy()

In [25]:
df

receta  \
0                  Iogurt grec amb fraules   
1                  Iogurt grec amb fraules   
2                  Iogurt grec amb fraules   
3                  Iogurt grec amb fraules   
4                  Iogurt grec amb fraules   
...                                    ...   
1058  Pollastre al forn a les fines herbes   
1059  Pollastre al forn a les fines herbes   
1060  Pollastre al forn a les fines herbes   
1061  Pollastre al forn a les fines herbes   
1062          Broqueta de gambes amb xoriç   

                    ingrediente_original  cantidad unidad ingrediente_estandar  
0                       Iogurt grec 200g       200      g          iogurt grec  
1                       Fraules 100-150g       150      g              fraules  
2                    Nous i ametlles 30g        30      g      nous i ametlles  
3                             Xía 10-20g        20      g                  xía  
4                       Canyella al gust         0    NaN             canyella  
...                                  ...       ...    ...                  ...  
1058    5g d'orenga fresc (0.5g en pols)         0    NaN         orenga fresc  
1059               15g de mostassa Dijon         0    NaN       mostassa dijon  
1060  15ml d'oli d'oliva (una cullerada)         0    NaN          oli d'oliva  
1061                 Sal i pebre al gust         0    NaN          sal i pebre  
1062                          llagostins       250      g           llagostins  

[1063 rows x 5 columns]

In [5]:
df.columns

Index(['receta', 'ingrediente_original', 'cantidad', 'unidad',
       'ingrediente_estandar'],
      dtype='object')

In [6]:
df.shape

(1063, 5)

In [7]:
df['ingrediente_estandar'].nunique()

231

### Líneas para manipular!

In [9]:
mask = df[(df['ingrediente_estandar'].str.contains("dents", na=False)) & (df['cantidad'] == 3)]

In [10]:
mask = df[df['ingrediente_estandar'].str.contains("essència", na=False)]

In [11]:
#df.loc[mask.index, 'ingrediente_estandar'] = 'essència de vainilla'
#df.loc[mask.index, 'cantidad'] = None
#df.loc[mask.index, 'unidad'] = None

In [12]:
df.loc[[802],'ingrediente_estandar'] = 'formatge'
#df.loc[[645],'cantidad'] = 1
#df.loc[[781],'unidad'] = None

In [13]:
#df[df['ingrediente_estandar'].str.contains("essència", na=False)]

n_fila = [{
    'receta': 'Broqueta de gambes amb xoriç',
    'ingrediente_original': 'llagostins',
    'cantidad': 250,
    'unidad': 'g',
    'ingrediente_estandar': 'llagostins'
}]

df = pd.concat([df, pd.DataFrame(n_fila)], ignore_index=True)

In [14]:
#df.sort_values(by=['ingrediente_estandar', 'cantidad'], ascending=[True, True])

In [15]:
#df['ingrediente_estandar'] = df['ingrediente_estandar'].replace('salmo fresc', 'salmó fresc')
#df['ingrediente_estandar'] = df['ingrediente_estandar'].replace('romani fresc', 'romaní fresc')


In [26]:
df.to_csv('ingredientes_estandarizados.csv', index=False)